README.md

```diff
@@ -187,6 +187,35 @@ if (idxs != null && idxs.length > 0) {
 uFuzzy provides a `uf.search(haystack, needle, outOfOrder = 0, infoThresh = 1e3) => [idxs, info, order]` wrapper which combines the `filter`, `info`, `sort` steps above.
 This method also implements efficient logic for matching search terms out of order and support for multiple substring exclusions, e.g. `fruit -green -melon`.
 
+Example of retrieving search results in order of ranking:
+
+```js
+let haystack = [
+  'example',
+  'exemple',
+  'another example',
+  'nonmatch'
+];
+
+let needle = 'example';
+
+let u = new uFuzzy({ intraMode: 1 });
+let [idxs,info,order] = u.search(haystack, needle);
+
+let sortHaystackByOrder = (idxs, order) => {
+  // Order may be null when doing out of order searching
+  // when the # of search terms exceed the max allowed 
```

Comment:
actually, any of them can be null.

iirs :rofl: ...

- `idxs` can be null if the search string ends up empty after trimming whitespace and removing non-alphanumeric chars (except those in quotes, for exact matches)
- `info` will be null if the number of results exceeds `infoThresh`

Reply:
and they're null in order.

if order is not null, then idxs and info is not null.
if info is not null then idxs is not null
if idxs is null, everything is null